pkg/stringutil/utils.go

```diff
@@ -74,3 +74,43 @@ func MakePathPrefixer(prefix string) func(string) string {
 		return prefix + path
 	}
 }
+
+// Test if a "s" string match a "p" pattern with joker (*, ?)
+func MatchWithJokers(p string, s string) bool {
```

Comment:
Let's use the more common term "wildcard" instead of "joker"

I'd prefer this function to have a couple table-driven tests to cover matching and non-matching scenarios for edge cases, such as:

* pattern contains `*`, `?`, `**`, `??`, `*?`, `?*`
* pattern starts with `*`, `?`, `**`, `??`, `*?`, `?*`
* pattern ends with `*`, `?`, `**`, `??`, `*?`, `?*`
* empty strings for pattern and string to be matched